src/main/java/org/opentripplanner/index/IndexGraphQLSchema.java

```diff
@@ -3054,6 +3055,55 @@ private Object getObject(String idString) {
                             }
                         })
                         .build())
+                .field(GraphQLFieldDefinition.newFieldDefinition()
+                        .name("cancelledTripTimes")
+                        .argument(GraphQLArgument.newArgument()
+                                .name("feedId")
+                                .description("ID of the feed.")
+                                .type(new GraphQLNonNull(Scalars.GraphQLString))
+                                .build())
+                        .argument(GraphQLArgument.newArgument()
+                                .name("serviceDate")
+                                .description("Date for which the TripTimes are returned. Format: YYYYMMDD. Default: returns all cancelled TripTimes.")
+                                .type(Scalars.GraphQLString)
+                                .build())
+                        .description("Get cancelled TripTimes.")
+                        .type(new GraphQLList(stoptimeType))
+                        .dataFetcher(environment -> {
+                            final String feedId = environment.getArgument("feedId");
+                            ServiceDate maybeServiceDate = null;
+                            try {
+                                maybeServiceDate = ServiceDate.parseString(environment.getArgument("serviceDate"));
+                            } catch (Exception e) {
+
+                            }
+                            final ServiceDate serviceDate = maybeServiceDate;
```

Comment:
3080 - 3104 should probably be a seperate method in GraphIndex

Reply:
Good point.